### PR TITLE
Re-send user payload if previously sent envelope wasn't acknowledged

### PR DESCRIPTION
**Pull request by dshulyak**
There are few changes merged into this PR. In the first one i removed debug API. It had only one method postSync, which allowed to wait for an envelope to get confirmed or expired. But it got into conflict with re-delivery approach and the only place where this API was used is in status-scale tests (they got abandoned anyway). So, i removed that code, if someone wants i can push those changes as a separate PR.

As for the re-delivery itself: when shhext_post called we will track user payload and original envelope ID. If that envelope expires before it get acknowledged, we will receive such event, and create new envelope to re-deliver payload. Signal that envelope expired will be sent only if we won't be able to deliver user payload with all attempts.

Maximum number of attempts can be configured with ShhextConfig.MaxMessageDeliveryAttempts options. Decided to go with single config option, but if we want different configuration per different message types we will have to add another API method. 

Envelope ID will be changing for every new envelope, cause it depends on the time when this envelope was sealed. But we need some unique way to distinguish a message, i used same approach as in our de-duplicator. I hashed the payload and the topic, and used that hash as a message ID. This message ID will be returned on shhext_post instead of envelope ID, and will be used in a signal to confirm message delivery or expiry. 

Closes # https://github.com/status-im/status-go/issues/1381

### CODE REVIEW

File: services/shhext/envelopes.go [hunk 1]
```diff
@@ -10,15 +12,46 @@ import (
 	whisper "github.com/status-im/whisper/whisperv6"
 )
 
+func messageID(message whisper.NewMessage) common.Hash {
```
Comment (mandrigin) [inline services/shhext/envelopes.go:15]: Let’s make it a separate type, otherwise maps look confusing. 

`type messageID common.Hash`

**Review by mandrigin — CHANGES_REQUESTED** [review]

**Review by pedropombeiro — APPROVED** [review]
lgtm, once Igor's suggestion is implemented

File: services/shhext/envelopes.go (hunk 1 above)
Comment (dshulyak) [reply services/shhext/envelopes.go:15]: not sure if it worth it. i need to cast it in several place to common.Hash, and copy String() method. and i am using it only for messageToEnvelope map

i will add a comment which key is used for each map

Comment (mandrigin) [reply services/shhext/envelopes.go:15]: okay

**Review by mandrigin — APPROVED** [review]

**Pull request merged**